### ISSUE TITLE
Running YoloV5's RTMP stream detection on an EC2 machine

### ISSUE THREAD

**Issue by nandukalidindi** [ultralytics/yolov5#4366]
## ❔Question
I am successfully able to run the detection script on an RTMP stream where the output is opened in a Python application window. However, when I tried to run the script on an EC2 micro instance (running Ubuntu). I get the below error.

```
Fusing layers... 
Model Summary: 224 layers, 7053910 parameters, 0 gradients, 16.3 GFLOPs
qt.qpa.xcb: could not connect to display :1.0
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/home/ubuntu/.local/lib/python3.8/site-packages/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb.

Aborted (core dumped)
```

I also tried commenting out these lines in the `detect.py` script as that is responsible for opening the application window (confirmed this by running on my machine and the application window does not show up).

```
# Stream results
if view_img:
    cv2.imshow(str(p), im0)
    cv2.waitKey(1)  # 1 millisecond
```

https://github.com/ultralytics/yolov5/blob/master/detect.py#L167

OS: `Linux Ubuntu SMP Thu Jul 15 11:07:29 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`

Image detection works as expected. Video detection runs into the same error as above.

If anyone has faced the issue or something similar, please do help. Thanks in advance.

**Comment by glenn-jocher** [comment]
@nandukalidindi EC2 questions should be directed to AWS support.

**Comment by cvanlit-supplai** [comment]
@nandukalidindi Had the same problem while running `python detect.py` from tmux, running it from a "normal" terminal worked. 

@glenn-jocher I'd like to be able to run it from tmux again (which worked before) so would be nice if this was fixed.

**Comment by nandukalidindi** [comment]
Yeah, I also ran the detection script in a docker container, the output warns that `cv2.imShow` is not supported in Docker environments but the execution goes forward by printing the detections. Is there any setting that we can enable to mimic the Docker container behaviour to allow for avoiding the display related errors as I am only interested in getting the printed detections, don't need an application window with the live stream.

**Comment by glenn-jocher** [comment]
@cvanlit-supplai @nandukalidindi 👋 hi, thanks for letting us know about this problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to get started investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

* ✅ **Minimal** – Use as little code as possible that still produces the same problem
* ✅ **Complete** – Provide **all** parts someone else needs to reproduce your problem in the question itself
* ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

In addition to the above requirements, for [Ultralytics](https://ultralytics.com/) to provide assistance your code should be:

* ✅ **Current** – Verify that your code is up-to-date with current GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been resolved by previous commits.
* ✅ **Unmodified** – Your problem must be reproducible without any modifications to the codebase in this repository. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all of the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) and providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by nandukalidindi** [comment]
@glenn-jocher I think the problem is with the EC2 `t2.micro` instance itself. It is not designed for intensive operations. Although I ran the detection script inside a docker container, the memory was just not enough. I did not try running the detection script directly on a more powerful machine. Either way, I saw that the docker container is the way to go with a relatively powerful EC2 instance (~p2.large) as suggested in the docs []((https://docs.ultralytics.com/environments/AWS-Quickstart/))

Closing this issue for now as the initial ask is based on insufficient software / hardware instance.
